docs/griptape-framework/structures/src/drivers_config_with.py

```diff
@@ -0,0 +1,31 @@
+import schema
+
+from griptape.configs.drivers import AnthropicDriversConfig, OpenAiDriversConfig
+from griptape.drivers import AnthropicPromptDriver, OpenAiChatPromptDriver
+from griptape.engines import JsonExtractionEngine
+from griptape.structures import Agent
+from griptape.tasks import ToolTask
+from griptape.tools import ExtractionTool
+
+with OpenAiDriversConfig():  # Agent will be created with OpenAi Drivers
+    openai_agent = Agent()
```

Comment:
Can you illustrate what is expected for this common scenario:

```py
def create_anthropic_agent():
    with AnthropicDriversConfig():
        return Agent()

anthropic_agent = create_anthropic_agent()

# Are there (and will there never be) any lazily instantiated objects
# inside Agent that could cause this to not use anthropic drivers?
anthropic_agent.run() 
```

Reply:
> Are there (and will there never be) any lazily instantiated objects inside Agent that could cause this to not use anthropic drivers?
Shouldn't be -- we don't lazily instantiate any objects in Structures.